modules/page/html/render/tag.js

```diff
@@ -0,0 +1,82 @@
+const { Value, computed, map, when, h } = require('mutant')
+var nest = require('depnest')
+
+exports.needs = nest({
+  'about.obs.valueFrom': 'first',
+  'keys.sync.id': 'first',
+  'message.html.render': 'first',
+  'message.obs.get': 'first',
+  'tag.html.tag': 'first',
+  'tag.obs.allTagsFrom': 'first',
+  'tag.obs.taggedMessages': 'first'
+})
+
+exports.gives = nest('page.html.render')
+
+exports.create = function (api) {
+  return nest('page.html.render', function channel (path) {
+    if (!path.startsWith('/tags')) return
+
+    const myId = api.keys.sync.id()
+    const tags = map(
+      api.tag.obs.allTagsFrom(myId),
+      tagId => {
+        return { tagId, tagName: api.about.obs.valueFrom(tagId, 'name', myId) }
+      }
+    )
+    const selectedTagId = Value(path.split('/')[2])
+    const name = computed(
+      selectedTagId,
+      tagId => tagId
+        ? api.about.obs.valueFrom(tagId, 'name', myId)
+        : Value('Select A Tag')
+    )
+    const tagMessages = computed(
+      selectedTagId,
+      tagId => tagId
+        ? map(
+            api.tag.obs.taggedMessages(myId, tagId),
+            msgId => api.message.obs.get(msgId)
+          )
+        : []
+    )
+
+    var prepend = [
+      h('PageHeading', [
+        h('h1', [ h('strong', 'Tags'), ' from you' ]),
+      ])
+    ]
+
+    return h('div.SplitView', [
+      h('div.side', [
+        h('h2', 'Your Tags'),
+        map(
+          tags,
+          tag => computed(
+            tag,
+            ({ tagId, tagName }) =>
+              h('a', {
+                'ev-click': () => selectedTagId.set(tagId),
```

Comment:
:warning: This means the page won't refresh, but it also means you're side-stepping the history, so that when people click back they won't get the last thing they were on.

I think you should push into the history (or however it's done here in patchwork). If you have problems with the view reloading being a jolt then you might need to build some caches, but figure that out later

Reply:
hmmm that's gonna change (and simplify) a bunch of your logic removing that observeable